modules/monitoring-configuring-remote-write.adoc

```diff
@@ -0,0 +1,176 @@
+* You have installed the OpenShift CLI (`oc`).
+* You have set up a remote write compatible endpoint (such as Thanos) and know the endpoint URL.
+* You have set up authentication credentials for the remote write endpoint. 
++
+[CAUTION]
+====
+To reduce security risks, avoid sending metrics to an endpoint via unencrypted HTTP or without using authentication.
+====
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add a `remoteWrite:` section under `data/config.yaml/prometheusK8s`.
+
+. In this section, add an endpoint URL, authetication credentials for the endpoint, and write label configurations.
+
+.. Add an endpoint URL and authentication credentials:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        <endpoint_authentication_credentials>
+----
++
+For `endpoint_authentication_credentials` substitute your credentials.
+Currently supported authentication methods are basic authentication (`basicAuth`) and client TLS (`tlsConfig`) authentication.
++
+* The following example configures basic authentication:
++
+[source,yaml]
+----
+        basicAuth:
+          username:
+            <usernameSecret>
+          password:
+            <passwordSecret>
+----
+Substitute `<usernameSecret>` and `<passwordSecret>` accordingly.
++
+* The following example configures client TLS authentication:
++
+[source,yaml]
+----
+         tlsConfig:
+           cert:
+              <certSecret>
+           keySecret:
+              <keySecret>
+----
+Substitute `<certSecret>` and `<keySecret>` accordingly.
++
+.. Add write relabel configuration values after the authentication credentials:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        <endpoint_authentication_credentials>
+        <write_relabel_configs>
+----
++
+For `<write_relabel_configs>` substitute a list of write relabel configurations for metrics that you want to send to the remote endpoint.
++
+See the link:https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config[Prometheus relabel_config documentation] for information about write relabel configuration options.
++
+The following example `cluster-monitoring-config` `ConfigMap` settings show how to:
++
+* forward a single metric called `my_metric`
++
+* configure basic authentication using the `remoteWriteAuth` secret with the `user` and `password` keys, which would contain the endpoint authentication credentials
```

Comment:
Did we just mix auth and rewire label config examples? Could these be split for reader clarity?

Reply:
This issue should be cleared up after I restructure the procedure.

Reply:
Updated. Thanks.